apps/ui/lib/components/CursorHOC.tsx

```diff
@@ -0,0 +1,141 @@
+import { Contract } from '@balena/jellyfish-types/build/core';
+import clone from 'deep-copy';
+import { circularDeepEqual } from 'fast-equals';
+import _ from 'lodash';
+import * as React from 'react';
+import { sdk } from '../core';
+
+interface Props {
+	query: Parameters<typeof sdk.getCursor>[0];
+	options: Parameters<typeof sdk.getCursor>[1];
+}
+
+interface State {
+	results: Contract[] | null;
+}
+
+export default class CursorHOC extends React.Component<Props, State> {
```

Comment:
If you create `useCursor` hook, you can use it inside `CursorHOC`, this way you have both HOC and hook with minimal code.

Reply:
Also, I think that this component is not HoC. See: https://reactjs.org/docs/higher-order-components.html

Reply:
I think I'll refactor this PR to put the filters, lens selection and cursor inside a new component. This way it will be waaay easier to drop in the whole thing as a block anywhere we render a collection (like the link tabs)